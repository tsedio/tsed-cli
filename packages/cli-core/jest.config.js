// For a detailed explanation regarding each configuration property, visit:
// https://jestjs.io/docs/en/configuration.html

module.exports = {
  ...require("@tsed/jest-config"),
  roots: ["<rootDir>/src", "<rootDir>/test"],
  moduleNameMapper: {
    "@tsed/cli-core": "<rootDir>/src/index.ts"
  },
  coverageThreshold: {
    global: {
      statements: 71.34,
      branches: 72.93,
      functions: 49.63,
      lines: 71.34
    }
  },
};
