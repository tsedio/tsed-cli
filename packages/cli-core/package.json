{
  "name": "@tsed/cli-core",
  "version": "3.20.17",
  "description": "Build your CLI with TypeScript and Decorators",
  "source": "./src/index.ts",
  "main": "./lib/index.js",
  "typings": "./lib/index.d.ts",
  "exports": {
    "require": "./lib/index.js",
    "default": "./lib/index.modern.js"
  },
  "keywords": [
    "cli",
    "typescript",
    "framework",
    "api",
    "decorators",
    "commander.js",
    "inquirer.js",
    "listr"
  ],
  "scripts": {
    "build": "microbundle --target node --no-compress --format modern,cjs --tsconfig ./tsconfig.compile.json"
  },
  "dependencies": {
    "@npmcli/run-script": "3.0.1",
    "@samverschueren/stream-to-observable": "^0.3.1",
    "@tsed/logger": ">=6.1.1",
    "any-observable": "0.5.1",
    "axios": "0.26.1",
    "chalk": "4.1.0",
    "change-case": "^4.1.2",
    "commander": "9.3.0",
    "consolidate": "0.16.0",
    "create-frame": "1.0.0",
    "enquirer": "2.3.6",
    "execa": "5.1.1",
    "figures": "4.0.0",
    "fs-extra": "10.1.0",
    "globby": "11.1.0",
    "handlebars": "4.7.7",
    "handlebars-utils": "1.0.6",
    "inquirer": "8.2.1",
    "inquirer-autocomplete-prompt": "2.0.0",
    "js-yaml": "4.1.0",
    "listr2": "4.0.5",
    "module-alias": "2.2.2",
    "normalize-path": "3.0.0",
    "read-pkg-up": "7.0.1",
    "registry-url": "5.1.0",
    "rxjs": "^7.5.5",
    "semver": "7.3.5",
    "split": "1.0.1",
    "tslib": "2.3.1",
    "tunnel": "0.0.6",
    "update-notifier": "5.1.0",
    "url-parse": "1.5.10"
  },
  "devDependencies": {
    "@types/axios": "0.14.0",
    "@types/commander": "2.12.2",
    "@types/consolidate": "0.14.1",
    "@types/figures": "3.0.1",
    "@types/fs-extra": "9.0.13",
    "@types/globby": "9.1.0",
    "@types/inquirer": "8.2.0",
    "@types/js-yaml": "4.0.5",
    "@types/normalize-path": "3.0.0",
    "@types/read-pkg-up": "6.0.0",
    "@types/semver": "7.3.9",
    "@types/tunnel": "0.0.3",
    "@types/update-notifier": "5.1.0",
    "@types/url-parse": "^1.4.3"
  },
  "peerDependencies": {
    "@tsed/core": ">=6.95.1",
    "@tsed/di": ">=6.95.1"
  }
}