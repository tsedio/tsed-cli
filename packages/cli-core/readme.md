# @tsed/cli-core

<p style="text-align: center" align="center">
 <a href="https://tsed.dev" target="_blank"><img src="https://tsed.dev/tsed-og.png" width="200" alt="Ts.ED logo"/></a>
</p>

[![Build & Release](https://github.com/tsedio/tsed-cli/workflows/Build%20&%20Release/badge.svg?branch=master)](https://github.com/tsedio/tsed-cli/actions?query=workflow%3A%22Build+%26+Release%22)
[![TypeScript](https://badges.frapsoft.com/typescript/love/typescript.svg?v=100)](https://github.com/ellerbrock/typescript-badges/)
[![Package Quality](http://npm.packagequality.com/shield/@tsed/cli.png)](http://packagequality.com/#?package=@tsed/cli)
[![npm version](https://badge.fury.io/js/%40tsed%2Fcli.svg)](https://badge.fury.io/js/%40tsed%2Fcli)
[![Dependencies](https://david-dm.org/tsedio/tsed-cli.svg)](https://david-dm.org/tsedio/tsed-cli#info=dependencies)
[![img](https://david-dm.org/tsedio/tsed-cli/dev-status.svg)](https://david-dm.org/tsedio/tsed-cli/#info=devDependencies)
[![img](https://david-dm.org/tsedio/tsed-cli/peer-status.svg)](https://david-dm.org/tsedio/tsed-cli/#info=peerDependenciess)
[![Known Vulnerabilities](https://snyk.io/test/github/tsedio/tsed-cli/badge.svg)](https://snyk.io/test/github/tsedio/ts-express-decorators)

> Create your CLI with TypeScript and decorators

## Goals

This package help TypeScript developers to build your own CLI with class and decorators. To doing that, @tsed/cli-core
use the Ts.ED DI and his utils to declare a new Command via decorators.

@tsed/cli-core provide also a plugin ready architecture. You and your community will be able to develop your official
cli-plugin and deploy it on npm registry.

## Features

- DI Framework (injection, configuration, etc...),
- Decorators,
- Extensible with plugins architecture.

Please refer to the [documentation](https://cli.tsed.dev/) for more details.

## Installation

```bash
npm install @tsed/core @tsed/di @tsed/cli-core
```

## Getting started

Create CLI require some steps like create a package.json with the right information and create a structure directory
aligned with TypeScript to be compiled correctly for a npm deployment.

Here a structure directory example:

```
.
‚îú‚îÄ‚îÄ lib -- Transpiled code
‚îú‚îÄ‚îÄ src -- TypeScript source
‚îÇ   ‚îú‚îÄ‚îÄ bin -- binary
‚îÇ   ‚îú‚îÄ‚îÄ commands
‚îÇ   ‚îî‚îÄ‚îÄ index.ts
‚îú‚îÄ‚îÄ templates -- Template files
‚îú‚îÄ‚îÄ package.json
‚îú‚îÄ‚îÄ tsconfig.json
‚îî‚îÄ‚îÄ tsconfig.compile.json
```

## Create package.json and tsconfig

The first step is to create the package.json with the following lines:

```jsonc
{
  "name": "{{name}}",
  "version": "1.0.0",
  "main": "./lib/index.js",
  "typings": "./lib/index.d.ts",
  "bin": {
    "tsed": "lib/bin/{{name}}.js"
  },
  "files": ["lib/bin/{{name}}.js", "lib/bin", "lib", "templates"],
  "description": "An awesome CLI build on top of @tsed/cli-core",
  "dependencies": {
    "@tsed/cli-core": "1.3.1",
    "tslib": "1.11.1"
  },
  "devDependencies": {
    "@tsed/cli-testing": "1.3.1",
    "ts-node": "latest",
    "typescript": "latest"
  },
  "scripts": {
    "build": "tsc --build tsconfig.compile.json",
    "start:cmd:add": "cross-env NODE_ENV=development ts-node -r src/bin/{{name}}.ts add -r ./.tmp"
  },
  "engines": {
    "node": ">=8.9"
  },
  "peerDependencies": {}
}
```

Then create tsconfig files one for the IDE (`tsconfig.json`):

```json
{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es2016",
    "sourceMap": true,
    "declaration": false,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "moduleResolution": "node",
    "isolatedModules": false,
    "suppressImplicitAnyIndexErrors": false,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "allowSyntheticDefaultImports": true,
    "importHelpers": true,
    "newLine": "LF",
    "noEmit": true,
    "lib": ["es7", "dom", "esnext.asynciterable"],
    "typeRoots": ["./node_modules/@types"]
  },
  "linterOptions": {
    "exclude": []
  },
  "exclude": []
}
```

And another one to compile source (`tsconfig.compile.json`):

```json
{
  "extends": "./tsconfig.compile.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "lib",
    "moduleResolution": "node",
    "declaration": true,
    "noResolve": false,
    "preserveConstEnums": true,
    "sourceMap": true,
    "noEmit": false,
    "inlineSources": true
  },
  "exclude": ["node_modules", "test", "lib", "**/*.spec.ts"]
}
```

## Create the bin file

The bin file is used by npm to create your node.js executable program when you install the node_module globally.

Create a new file according to your project name (example: `name.ts`) and add this code:

```typescript
#!/usr/bin/env node
import {AddCmd, CliCore} from "@tsed/cli-core";
import {resolve} from "node:path";

const pkg = require("../../package.json");
const TEMPLATE_DIR = resolve(__dirname, "..", "..", "templates");

CliCore.bootstrap({
  commands: [
    AddCmd // CommandProvider to install a plugin
    // then add you commands
  ],

  // optionals
  name: "name", // replace by the cli name. This property will be used by Plugins command
  pkg,
  templateDir: TEMPLATE_DIR
}).catch(console.error);
```

## Create your first command

```typescript
import {Command, CommandProvider, ClassNamePipe, OutputFilePathPipe, Inject, RoutePipe, SrcRendererService} from "@tsed/cli-core";

export interface GenerateCmdContext {
  type: string;
  name: string;
}

@Command({
  name: "generate",
  description: "Generate a new provider class",
  args: {
    type: {
      description: "Type of the provider (Injectable, Controller, Pipe, etc...)",
      type: String
    },
    name: {
      description: "Name of the class",
      type: String
    }
  }
})
export class GenerateCmd implements CommandProvider {
  @Inject()
  classNamePipe: ClassNamePipe;

  @Inject()
  outputFilePathPipe: OutputFilePathPipe;

  @Inject()
  routePipe: RoutePipe;

  @Inject()
  srcRenderService: SrcRendererService;

  /**
   * Prompt use Inquirer.js to print questions (see Inquirer.js for more details)
   */
  $prompt(initialOptions: Partial<GenerateCmdContext>) {
    return [
      {
        type: "list",
        name: "type",
        message: "Which type of provider ?",
        default: initialOptions.type,
        when: !initialOptions.type,
        choices: ["injectable", "decorator"]
      },
      {
        type: "input",
        name: "name",
        message: "Which name ?",
        when: !initialOptions.name
      }
    ];
  }

  /**
   * Map context is called before $exec and map use answers.
   * This context will be given for your $exec method and will be forwarded to other plugins
   */
  $mapContext(ctx: Partial<GenerateCmdContext>): GenerateCmdContext {
    const {name = "", type = ""} = ctx;

    return {
      ...ctx,
      symbolName: this.classNamePipe.transform({name, type}),
      outputFile: `${this.outputFilePathPipe.transform({name, type})}.ts`
    } as IGenerateCmdContext;
  }

  /**
   * Perform action like generate files. The tasks returned by $exec method is based on Listr configuration (see Listr documentation on npm)
   */
  async $exec(options: GenerateCmdContext) {
    const {outputFile, ...data} = options;

    const template = `generate/${options.type}.hbs`;

    return [
      {
        title: `Generate ${options.type} file to '${outputFile}'`,
        task: () =>
          this.srcRenderService.render(template, data, {
            output: outputFile
          })
      }
    ];
  }
}
```

Finally, create a handlebars template in templates directory:

```hbs
import {Injectable} from "@tsed/di"; @Injectable() export class {{symbolName}} { }
```

## Run command in dev mode

In your package.json add the following line in scripts property:

```
{
  "start:cmd:generate": "cross-env NODE_ENV=development ts-node -r src/bin/{{name}}.ts generate -r ./.tmp"
}
```

> Note: replace {{name}} by the name of you bin file located in src/bin.

> Note 2: The option `-r ./.tmp` create a temporary directory to generate files with your command.

## More examples

Here other commands examples:

- Init a project command: https://github.com/tsedio/tsed-cli/tree/master/packages/cli/src/commands/init/InitCmd
- Generate command: https://github.com/tsedio/tsed-cli/tree/master/packages/cli/src/commands/generate/GenerateCmd
- Plugin example: https://github.com/tsedio/tsed-cli/tree/master/packages/cli-plugin-mocha
- Mono repo CLI: https://github.com/tsedio/tsed-cli/tree/master

## Contributors

Please read [contributing guidelines here](https://tsed.dev/CONTRIBUTING.html)

<a href="https://github.com/tsedio/ts-express-decorators/graphs/contributors"><img src="https://opencollective.com/tsed/contributors.svg?width=890" /></a>

## Backers

Thank you to all our backers! üôè [[Become a backer](https://opencollective.com/tsed#backer)]

<a href="https://opencollective.com/tsed#backers" target="_blank"><img src="https://opencollective.com/tsed/tiers/backer.svg?width=890"></a>

## Sponsors

Support this project by becoming a sponsor. Your logo will show up here with a link to your
website. [[Become a sponsor](https://opencollective.com/tsed#sponsor)]

## License

The MIT License (MIT)

Copyright (c) 2016 - 2023 Romain Lenzotti

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
