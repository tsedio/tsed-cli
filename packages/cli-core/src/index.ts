import "./utils/patchCommander.js";

import Inquirer from "inquirer";

export * from "./CliCore.js";
export * from "./decorators/index.js";
export * from "./interfaces/index.js";
export * from "./packageManagers/index.js";
export * from "./services/index.js";
export * from "./utils/index.js";
export * from "@tsed/core";
export * from "@tsed/di";
export * from "@tsed/logger";
export * from "@tsed/normalize-path";
export {Inquirer};
