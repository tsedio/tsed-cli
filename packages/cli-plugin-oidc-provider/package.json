{
  "name": "@tsed/cli-plugin-oidc-provider",
  "description": "Ts.ED CLI plugin. Add OIDC Provider",
  "version": "6.4.4",
  "type": "module",
  "main": "./lib/esm/index.js",
  "source": "./src/index.ts",
  "module": "./lib/esm/index.js",
  "typings": "./lib/types/index.d.ts",
  "exports": {
    ".": {
      "cli-tsed-source": "./src/index.ts",
      "types": "./lib/types/index.d.ts",
      "import": "./lib/esm/index.js",
      "default": "./lib/esm/index.js"
    }
  },
  "scripts": {
    "build": "yarn build:ts",
    "build:ts": "tsc --build tsconfig.esm.json",
    "test": "vitest run",
    "test:ci": "vitest run --coverage.thresholds.autoUpdate=true"
  },
  "files": [
    "lib",
    "templates"
  ],
  "dependencies": {
    "tslib": "2.7.0"
  },
  "devDependencies": {
    "@tsed/cli": "workspace:*",
    "@tsed/cli-core": "workspace:*",
    "@tsed/typescript": "workspace:*",
    "cross-env": "7.0.3",
    "typescript": "5.6.2",
    "vitest": "3.2.4"
  },
  "peerDependencies": {
    "@tsed/cli": "workspace:*",
    "@tsed/cli-core": "workspace:*"
  }
}
