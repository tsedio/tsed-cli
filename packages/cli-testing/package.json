{
  "name": "@tsed/cli-testing",
  "description": "Utils to test you CLI based on Ts.ED CLI",
  "version": "6.4.4",
  "type": "module",
  "main": "./lib/esm/index.js",
  "source": "./src/index.ts",
  "module": "./lib/esm/index.js",
  "typings": "./lib/types/index.d.ts",
  "exports": {
    ".": {
      "cli-tsed-source": "./src/index.ts",
      "types": "./lib/types/index.d.ts",
      "import": "./lib/esm/index.js",
      "default": "./lib/esm/index.js"
    }
  },
  "scripts": {
    "build": "yarn build:ts",
    "build:ts": "tsc --build tsconfig.esm.json"
  },
  "keywords": [
    "Ts.ED",
    "cli",
    "typescript",
    "framework",
    "rest",
    "api",
    "express",
    "decorators"
  ],
  "dependencies": {
    "@tsed/cli-core": "workspace:*",
    "tslib": "2.7.0"
  },
  "devDependencies": {
    "@tsed/typescript": "workspace:*",
    "cross-env": "7.0.3",
    "typescript": "5.6.2",
    "vitest": "3.2.4"
  }
}
