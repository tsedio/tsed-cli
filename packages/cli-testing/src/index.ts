export * from "./CliPlatformTest.js";
export * from "./FakeCliExeca.js";
export * from "./FakeCliFs.js";
export * from "./FakeCliHttpClient.js";
export * from "./normalizePath.js";
