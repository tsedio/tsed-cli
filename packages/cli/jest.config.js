// For a detailed explanation regarding each configuration property, visit:
// https://jestjs.io/docs/en/configuration.html

module.exports = {
  ...require("@tsed/jest-config"),
  roots: ["<rootDir>/src", "<rootDir>/test"],

  coverageThreshold: {
    global: {
      statements: 93.01,
      branches: 80.73,
      functions: 83.47,
      lines: 93.01
    }
  }
};
