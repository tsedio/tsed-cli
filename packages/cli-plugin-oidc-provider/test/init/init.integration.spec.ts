import {CliService, ProjectPackageJson} from "@tsed/cli-core";
import {CliPlatformTest, FakeCliFs} from "@tsed/cli-testing";
import {ensureDirSync, existsSync, readFileSync, writeFileSync} from "fs-extra";
import {InitCmd, TEMPLATE_DIR} from "@tsed/cli";
import {dirname} from "path";
import "@tsed/cli-plugin-oidc-provider";

function readFile(file: string, content: string, rewrite = false) {
  const path = `${__dirname}/${file}`

  ensureDirSync(dirname(path))

  if (!existsSync(path) || rewrite) {
    writeFileSync(path, content, {encoding: "utf8"});
  }

  return readFileSync(path, {encoding: "utf8"});
}

describe("Init OIDC Provider project", () => {
  beforeEach(() =>
    CliPlatformTest.bootstrap({
      templateDir: TEMPLATE_DIR,
      commands: [InitCmd]
    })
  );
  afterEach(() => CliPlatformTest.reset());

  it("should generate a project with oidc", async () => {
    const cliService = CliPlatformTest.get<CliService>(CliService);
    const projectPackageJson = CliPlatformTest.get<ProjectPackageJson>(ProjectPackageJson);
    projectPackageJson.setRaw({
      name: "",
      version: "1.0.0",
      description: "",
      scripts: {},
      dependencies: {},
      devDependencies: {}
    });

    await cliService.exec("init", {
      platform: "express",
      rootDir: "./project-data",
      projectName: "project-data",
      tsedVersion: "5.58.1",
      oidc: true,
      oidcBasePath: "/oidc"
    });

    expect(FakeCliFs.getKeys()).toEqual([
      "./project-name",
      "project-name",
      "project-name/.barrelsby.json",
      "project-name/.dockerignore",
      "project-name/.gitignore",
      "project-name/Dockerfile",
      "project-name/README.md",
      "project-name/docker-compose.yml",
      "project-name/package.json",
      "project-name/processes.config.js",
      "project-name/src",
      "project-name/src/Server.ts",
      "project-name/src/config",
      "project-name/src/config/envs",
      "project-name/src/config/envs/index.ts",
      "project-name/src/config/index.ts",
      "project-name/src/config/logger",
      "project-name/src/config/logger/index.ts",
      "project-name/src/config/oidc",
      "project-name/src/config/oidc/index.ts",
      "project-name/src/controllers/oidc",
      "project-name/src/controllers/oidc/InteractionsController.ts",
      "project-name/src/controllers/rest",
      "project-name/src/controllers/rest/HelloWorldController.ts",
      "project-name/src/index.ts",
      "project-name/src/interactions",
      "project-name/src/interactions/ConsentInteraction.ts",
      "project-name/src/interactions/CustomInteraction.ts",
      "project-name/src/interactions/LoginInteraction.ts",
      "project-name/src/models",
      "project-name/src/models/Account.ts",
      "project-name/src/services",
      "project-name/src/services/Accounts.ts",
      "project-name/tsconfig.compile.json",
      "project-name/tsconfig.json",
      "project-name/views",
      "project-name/views/forms",
      "project-name/views/forms/interaction-form.ejs",
      "project-name/views/forms/login-form.ejs",
      "project-name/views/forms/select-account-form.ejs",
      "project-name/views/interaction.ejs",
      "project-name/views/login.ejs",
      "project-name/views/partials",
      "project-name/views/partials/footer.ejs",
      "project-name/views/partials/header.ejs",
      "project-name/views/partials/login-help.ejs",
      "project-name/views/repost.ejs",
      "project-name/views/select_account.ejs"
    ]);

    const content = FakeCliFs.entries.get("project-name/src/Server.ts")!;

    expect(content).toContain("import \"@tsed/oidc-provider\"");
    expect(content).toContain("import {InteractionsController} from \"./controllers/oidc/InteractionsController\";");
    expect(content).toEqual(readFile("data/Server.express.oidc.ts.txt", content));

    const configContent = FakeCliFs.entries.get("project-name/src/config/oidc/index.ts")!;

    expect(configContent).toContain("path: \"/oidc\"");
  });
  it("should generate a project with oidc and swagger", async () => {
    const cliService = CliPlatformTest.get<CliService>(CliService);
    const projectPackageJson = CliPlatformTest.get<ProjectPackageJson>(ProjectPackageJson);
    projectPackageJson.setRaw({
      name: "",
      version: "1.0.0",
      description: "",
      scripts: {},
      dependencies: {},
      devDependencies: {}
    });

    await cliService.exec("init", {
      platform: "express",
      rootDir: "./project-data",
      projectName: "project-data",
      tsedVersion: "5.58.1",
      oidc: true,
      swagger: true,
      oidcBasePath: "/oidc"
    });

    expect(FakeCliFs.getKeys()).toEqual([
      "./project-name",
      "project-name",
      "project-name/.barrelsby.json",
      "project-name/.dockerignore",
      "project-name/.gitignore",
      "project-name/Dockerfile",
      "project-name/README.md",
      "project-name/docker-compose.yml",
      "project-name/package.json",
      "project-name/processes.config.js",
      "project-name/src",
      "project-name/src/Server.ts",
      "project-name/src/config",
      "project-name/src/config/envs",
      "project-name/src/config/envs/index.ts",
      "project-name/src/config/index.ts",
      "project-name/src/config/logger",
      "project-name/src/config/logger/index.ts",
      "project-name/src/config/oidc",
      "project-name/src/config/oidc/index.ts",
      "project-name/src/controllers/oidc",
      "project-name/src/controllers/oidc/InteractionsController.ts",
      "project-name/src/controllers/pages",
      "project-name/src/controllers/pages/IndexController.ts",
      "project-name/src/controllers/rest",
      "project-name/src/controllers/rest/HelloWorldController.ts",
      "project-name/src/index.ts",
      "project-name/src/interactions",
      "project-name/src/interactions/ConsentInteraction.ts",
      "project-name/src/interactions/CustomInteraction.ts",
      "project-name/src/interactions/LoginInteraction.ts",
      "project-name/src/models",
      "project-name/src/models/Account.ts",
      "project-name/src/services",
      "project-name/src/services/Accounts.ts",
      "project-name/tsconfig.compile.json",
      "project-name/tsconfig.json",
      "project-name/views",
      "project-name/views/forms",
      "project-name/views/forms/interaction-form.ejs",
      "project-name/views/forms/login-form.ejs",
      "project-name/views/forms/select-account-form.ejs",
      "project-name/views/interaction.ejs",
      "project-name/views/login.ejs",
      "project-name/views/partials",
      "project-name/views/partials/footer.ejs",
      "project-name/views/partials/header.ejs",
      "project-name/views/partials/login-help.ejs",
      "project-name/views/repost.ejs",
      "project-name/views/select_account.ejs",
      "project-name/views/swagger.ejs"
    ]);

    const content = FakeCliFs.entries.get("project-name/src/Server.ts")!;

    expect(content).toContain("import \"@tsed/oidc-provider\"");
    expect(content).toContain("import {InteractionsController} from \"./controllers/oidc/InteractionsController\";");
    expect(content).toEqual(readFile("data/Server.express.oidc.swagger.ts.txt", content));

    const configContent = FakeCliFs.entries.get("project-name/src/config/oidc/index.ts")!;

    expect(configContent).toContain("path: \"/oidc\"");
  });
});
