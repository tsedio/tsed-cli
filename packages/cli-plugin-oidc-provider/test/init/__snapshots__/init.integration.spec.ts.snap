// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Init OIDC Provider project > should generate a project with oidc 1`] = `
[
  "project-name",
  "project-name/.barrels.json",
  "project-name/.dockerignore",
  "project-name/.gitignore",
  "project-name/.swcrc",
  "project-name/Dockerfile",
  "project-name/README.md",
  "project-name/docker-compose.yml",
  "project-name/nodemon.json",
  "project-name/package.json",
  "project-name/processes.config.cjs",
  "project-name/src",
  "project-name/src/Server.ts",
  "project-name/src/config",
  "project-name/src/config/envs",
  "project-name/src/config/envs/index.ts",
  "project-name/src/config/index.ts",
  "project-name/src/config/logger",
  "project-name/src/config/logger/index.ts",
  "project-name/src/config/oidc",
  "project-name/src/config/oidc/index.ts",
  "project-name/src/controllers/oidc",
  "project-name/src/controllers/oidc/InteractionsController.spec.ts",
  "project-name/src/controllers/oidc/InteractionsController.ts",
  "project-name/src/controllers/rest",
  "project-name/src/controllers/rest/HelloWorldController.ts",
  "project-name/src/index.ts",
  "project-name/src/interactions",
  "project-name/src/interactions/ConsentInteraction.spec.ts",
  "project-name/src/interactions/ConsentInteraction.ts",
  "project-name/src/interactions/CustomInteraction.ts",
  "project-name/src/interactions/LoginInteraction.spec.ts",
  "project-name/src/interactions/LoginInteraction.ts",
  "project-name/src/interactions/__mock__",
  "project-name/src/interactions/__mock__/oidcContext.fixture.ts",
  "project-name/src/models",
  "project-name/src/models/Account.ts",
  "project-name/src/services",
  "project-name/src/services/Accounts.ts",
  "project-name/tsconfig.base.json",
  "project-name/tsconfig.json",
  "project-name/tsconfig.node.json",
  "project-name/views",
  "project-name/views/consent.ejs",
  "project-name/views/forms",
  "project-name/views/forms/consent-form.ejs",
  "project-name/views/forms/login-form.ejs",
  "project-name/views/forms/select-account-form.ejs",
  "project-name/views/login.ejs",
  "project-name/views/partials",
  "project-name/views/partials/footer.ejs",
  "project-name/views/partials/header.ejs",
  "project-name/views/partials/login-help.ejs",
  "project-name/views/repost.ejs",
  "project-name/views/select_account.ejs",
]
`;

exports[`Init OIDC Provider project > should generate a project with oidc 2`] = `
"import {join} from "node:path";
import {Configuration} from "@tsed/di";
import {application} from "@tsed/platform-http";
import "@tsed/platform-log-request"; // remove this import if you don&#x27;t want log request
import "@tsed/platform-express"; // /!\\ keep this import
import "@tsed/ajv";
import "@tsed/oidc-provider";
import {config} from "./config/index.js";
import * as rest from "./controllers/rest/index.js";
import * as pages from "./controllers/pages/index.js";
import {InteractionsController} from "./controllers/oidc/InteractionsController.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [
      ...Object.values(rest)
    ],
    "/": [
      InteractionsController
    ]
  },
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    { use: "urlencoded-parser", options: { extended: true }}
  ],
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  }
})
export class Server {
  protected app = application();
}
"
`;

exports[`Init OIDC Provider project > should generate a project with oidc and swagger 1`] = `
[
  "project-name",
  "project-name/.barrels.json",
  "project-name/.dockerignore",
  "project-name/.gitignore",
  "project-name/.swcrc",
  "project-name/Dockerfile",
  "project-name/README.md",
  "project-name/docker-compose.yml",
  "project-name/nodemon.json",
  "project-name/package.json",
  "project-name/processes.config.cjs",
  "project-name/src",
  "project-name/src/Server.ts",
  "project-name/src/config",
  "project-name/src/config/envs",
  "project-name/src/config/envs/index.ts",
  "project-name/src/config/index.ts",
  "project-name/src/config/logger",
  "project-name/src/config/logger/index.ts",
  "project-name/src/config/oidc",
  "project-name/src/config/oidc/index.ts",
  "project-name/src/controllers/oidc",
  "project-name/src/controllers/oidc/InteractionsController.ts",
  "project-name/src/controllers/pages",
  "project-name/src/controllers/pages/IndexController.ts",
  "project-name/src/controllers/rest",
  "project-name/src/controllers/rest/HelloWorldController.ts",
  "project-name/src/index.ts",
  "project-name/src/interactions",
  "project-name/src/interactions/ConsentInteraction.ts",
  "project-name/src/interactions/CustomInteraction.ts",
  "project-name/src/interactions/LoginInteraction.ts",
  "project-name/src/models",
  "project-name/src/models/Account.ts",
  "project-name/src/services",
  "project-name/src/services/Accounts.ts",
  "project-name/tsconfig.base.json",
  "project-name/tsconfig.json",
  "project-name/tsconfig.node.json",
  "project-name/views",
  "project-name/views/consent.ejs",
  "project-name/views/forms",
  "project-name/views/forms/consent-form.ejs",
  "project-name/views/forms/login-form.ejs",
  "project-name/views/forms/select-account-form.ejs",
  "project-name/views/login.ejs",
  "project-name/views/partials",
  "project-name/views/partials/footer.ejs",
  "project-name/views/partials/header.ejs",
  "project-name/views/partials/login-help.ejs",
  "project-name/views/repost.ejs",
  "project-name/views/select_account.ejs",
  "project-name/views/swagger.ejs",
]
`;

exports[`Init OIDC Provider project > should generate a project with oidc and swagger 2`] = `
"import {join} from "node:path";
import {Configuration} from "@tsed/di";
import {application} from "@tsed/platform-http";
import "@tsed/platform-log-request"; // remove this import if you don&#x27;t want log request
import "@tsed/platform-express"; // /!\\ keep this import
import "@tsed/ajv";
import "@tsed/swagger";
import "@tsed/oidc-provider";
import {config} from "./config/index.js";
import * as rest from "./controllers/rest/index.js";
import * as pages from "./controllers/pages/index.js";
import {InteractionsController} from "./controllers/oidc/InteractionsController.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [
      ...Object.values(rest)
    ],
    "/": [
      ...Object.values(pages),
      InteractionsController
    ]
  },
  swagger: [
    {
      path: "/doc",
      specVersion: "3.0.1"
    }
  ],
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    { use: "urlencoded-parser", options: { extended: true }}
  ],
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  }
})
export class Server {
  protected app = application();
}
"
`;
